python/lsst/daf/butler/core/registry.py

```diff
@@ -349,6 +360,17 @@ def getDatasetType(self, name):
         """
         raise NotImplementedError("Must be implemented by subclass")
 
+    @abstractmethod
+    def getAllDatasetTypes(self):
+        r"""Get every registered `DatasetType`.
+
+        Returns
+        -------
+        types : `tuple` of `DatasetType`
```

Comment:
Why is it a tuple? (I'm probably an old school guy where tuple means a "record", and list means a sequence).

Reply:
I don't really like returning mutable object for something that is meant to reflect a fixed state. I think of tuple as a read-only list.

Reply:
Could you return `frozenset` instead?

Reply:
Yes. Not sure why I don't.